### PR TITLE
linting and style for UI

### PR DESCRIPTION
**Pull request by rymurr**
This PR does:

* adds tslint checking - code quality/style
* prettier formatting
* adds prettier autoformating to pre-commit
* adds tslint/prettier to CI
* fixes existing linting and formatting

---
This change is [Reviewable](https://reviewable.io/reviews/projectnessie/nessie/1919)

### CODE REVIEW

**Review by omarsmak — APPROVED** [review]

**Review by dimas-b — DISMISSED** [review]

File: ui/src/utils/api/configuration.ts [hunk 1]
```diff
@@ -19,71 +19,72 @@
  * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
  *
  * OpenAPI spec version: 0.2.0
- * 
+ *
  *
  * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
  * https://openapi-generator.tech
  * Do not edit the class manually.
```
Comment (dimas-b) [inline ui/src/utils/api/configuration.ts:26]: Should it be under `.../generated`?

**Review by dimas-b — APPROVED** [review]

File: ui/src/utils/api/configuration.ts (hunk 1 above)
Comment (dimas-b) [reply ui/src/utils/api/configuration.ts:26]: Might be an old artifact that I missed when I introduced the `generated` directory for UI... is it still used?

**Review by omarsmak — DISMISSED** [review]

File: ui/src/utils/api/configuration.ts (hunk 1 above)
Comment (rymurr) [reply ui/src/utils/api/configuration.ts:26]: i just removed it. What about `custom.d.ts`? Is that still required too? That is in the same directory.

Comment (dimas-b) [reply ui/src/utils/api/configuration.ts]: TBH, I do not really know :)

**Review by dimas-b — APPROVED** [review]

File: ui/src/utils/api/configuration.ts (hunk 1 above)
Comment (rymurr) [reply ui/src/utils/api/configuration.ts]: i''ll leave it for now and test/remove later :-) thanks

**Pull request merged**